src/main/java/TaskList.java

```diff
@@ -0,0 +1,196 @@
+     */
+    public static String getDeadlineTask(String input) {
+        int dateIndex = input.indexOf("(by: ");
+        return input.substring(16, dateIndex - 1);
+    }
+
+    /**
+     * Returns deadline task date from filtering the input
+     * @param input the task line previously loaded into the file
+     * @return the deadline task date
+     */
+    public static String getDeadlineDate(String input) {
+        int dateIndex = input.indexOf("(by: ");
+        return input.substring(dateIndex + 5, dateIndex + 15);
+    }
+
+    /**
+     * Returns event task description from filtering the input
+     * @param input the task line previously loaded into the file
+     * @return the event task description
+     */
+    public static String getEventTask(String input) {
+        int dateIndex = input.indexOf("(at: ");
+        return input.substring(16, dateIndex - 1);
+    }
+
+    /**
+     * Returns event task date from filtering the input
+     * @param input the task line previously loaded into the file
+     * @return the event task date
+     */
+    public static String getEventDate(String input) {
+        int dateIndex = input.indexOf("(at: ");
+        return input.substring(dateIndex + 5, input.length() - 2);
+    }
+
+    /**
+     * Checks if the task stored in the file has been completed
+     * @param storedTaskString task stored in the file
+     */
+    public static void checkIfTaskDone(String storedTaskString) {
+        if (storedTaskString.substring(4).startsWith("C")) {
+            setDone(updatedTaskList.get(updatedTaskList.size() - 1));
+        }
+    }
+
+    /**
+     * Adds task into the task list
+     * @param task to be added into the task list
+     */
+    public static void addTaskToList(Task task) {
+        updatedTaskList.add(task);
+    }
+
+    /**
+     * Creates an event task and adds it to the task list
+     * @param eventTask event task description
+     * @param date of which the event occurs
+     */
+    public static void createEventTask(String eventTask, String date) {
+        Task task = new Event(eventTask, date);
+        addTaskToList(task);
+    }
+
+
+    /**
+     * Creates an deadline task and adds it to the task list
+     * @param deadlineTask deadline task description
+     * @param deadline of which the task is supposed to be completed by
+     */
+    public static void createDeadlineTask(String deadlineTask, LocalDate deadline) {
+        Task task = new Deadline(deadlineTask, deadline);
+        addTaskToList(task);
+    }
+
+    /**
+     * Creates a todo task and adds it to the task list
+     * @param todoTask todo task description
+     */
+    public static void createTodoTask(String todoTask) {
+        Task task = new Todo(todoTask);
+        addTaskToList(task);
+    }
+
+    /**
+     * Mark task as done
+     * @param task to be marked as done
+     */
+    public static void setDone(Task task) {
+        task.isDone = true;
```

Comment:
Perhaps instead of accessing a task attribute directly, you might want to consider using a setter method such as task.setTaskAsDone() to perform this operation